File: src/main/java/ch07/Ex07.java
package ch07;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

public class Ex07 {


	public static void main(String[] args) throws IOException, InterruptedException {
		ProcessBuilder pb = new ProcessBuilder("cmd", "/c", "ping");
		Process process = pb.start();

		int ret = process.waitFor();
		System.out.println("戻り値：" + ret);
		InputStream is = process.getInputStream();	//標準出力
		printInputStream(is);
	}

	public static void printInputStream(InputStream is) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(is));
		try {
			for (;;) {
				String line = br.readLine();
				if (line == null) break;
				System.out.println(line);
			}
		} finally {
			br.close();
		}
	}
}

File: src/main/java/ch07/ex07.js
ProcessBuilder = java.lang.ProcessBuilder
BufferedReader = java.io.BufferedReader;
InputStreamReader =  java.io.InputStreamReader;

function pipe() {
	if(arguments == null || arguments.length == 0){
		return;
	}
	var arg = new Array();
	arg.push("cmd");
	arg.push("/c");
	for(var i=0; i<arguments.length; i++){
		arg.push(arguments[i]);
	}
	var pb = new ProcessBuilder(arg);
	var process = pb.start();
	var ret = process.waitFor();
	print("戻り値：" + ret);
	printInputStream( process.getInputStream());
}

function printInputStream()  {
	var br = new BufferedReader(new InputStreamReader(arguments[0]));
	try {
		for (;;) {
			var line = br.readLine();
			if (line == null) break;
			print(line);
		}
	} finally {
		br.close();
	}
}

pipe('ping|find "a"');

